### PR TITLE
Add insecure_no_tls flag to the YAML config

### LINKED ISSUE #2147: Teleport Agent Command Line arguments should be available in teleport.yaml config - `insecure-no-tls`
I am not sure if this is consistent in the Teleport agent yet, but it would be good if all CLI arguments are available to configure in the `teleport.yaml`. One example encountered is when configuring `insecure-no-tls`

**What happened**:

Teleport complains `insecure-no-tls` config is not valid when including it in the YAML config.

**What you expected to happen**:

When `/usr/local/bin/teleport start --config=/etc/teleport.yaml` is run, it detects the `insecure-no-tls` config in the YAML config.

**How to reproduce it (as minimally and precisely as possible)**:

Include the configuration `insecure-no-tls` in teleport.yaml to disable TLS.

Run `/usr/local/bin/teleport start --config=/etc/teleport.yaml`

**Environment**:
- Teleport version (use `teleport version`): 2.7.1 (enterprise)
- Tsh version (use `tsh version`):
- OS (e.g. from /etc/os-release): Ubuntu 14.04

### PR DESCRIPTION
**Pull request by cd-champion-dev**
Fixes #2147

### CODE REVIEW

File: lib/config/configuration.go [hunk 1]
```diff
@@ -320,6 +320,11 @@ func ApplyFileConfig(fc *FileConfig, cfg *service.Config) error {
 	}
 	cfg.CachePolicy = *cachePolicy
 
+	// Apply insecure_no_tls
+	if (fc.DisableTLS != nil {
```
Comment (webvictim) [inline lib/config/configuration.go:324]: This bracket isn't closed?

Comment (d5c4b3) [reply lib/config/configuration.go]: Oops 😬. Fixed it.

**Comment by webvictim** [comment]
I think originally we didn't expose this in the YAML config to make it harder to use, because deploying without TLS should be a last resort and never used in production.

**Comment by d5c4b3** [comment]
The issue had a number of people that wanted to run teleport behind a reverse proxy that did TLS termination. My motivation for submitting the pull request was to help a friend that was struggling to to get haproxy and their firewall to play nicely with teleport.

**Comment by d5c4b3** [comment]
And by "the issue" I meant the one for the original disable-tls flag
https://github.com/gravitational/teleport/issues/1304

**Comment by Engrave-zz** [comment]
had this issue in the teleport-cluster helm chart as well but worked around it by adding the following, then i was able to successfully terminate ssl at aws alb and connect to the service:

```
extraArgs:
  - "--insecure-no-tls"
```

**Comment by programmerq** [comment]
I am of the opinion that the existing `--insecure-no-tls` functionality does not belong in Teleport at all. Even in a case where you have a reverse proxy terminating the TLS connection in front of teleport, it is still best practice to have encryption enabled everywhere.

I am against making it easier to disable encryption.

**Comment by r0mant** [comment]
@d5c4b3 Thank you for your contribution! We recommend providing this setting on the command line instead of specifying it in the configuration file.

**Pull request closed without merging**